File: tests/perf2/test-function-call.js
/*
 *  Test basic function call.
 */

if (typeof window == 'undefined') window = {}
if (!window.performance) window.performance = {}
window.performance.now = window.performance.now || Date.now

if (typeof print != 'undefined') log = print
else log = console.log

function f () { return }

function test () {
  var start
  var duration

  start = window.performance.now()

  // begin test
  for (var i = 0; i < 1e8; i++) f()
  // end test

  duration = window.performance.now() - start

  log(duration)
}

test()

File: tests/perf2/test-function-call.c
/*
 * Test basic function call.
 */

#include "stdio.h"
#include "time.h"

void f () { return; }

int main(int argc, char **argv) {
     int msec;
     clock_t start, duration;

     start = clock();

     // begin test
     for (int i = 0; i < 1e8; i++) f();
     // end test

     duration = clock() - start;
     msec = duration * 1000 / CLOCKS_PER_SEC;

     printf("%d\n", msec);

     return 0;
}
